commit 0752f78b9bc463d3155c2f2581ed07aa05834983
Author: MingZhu Yan <yanmingzhu@iscas.ac.cn>
Date:   Sun Nov 10 09:47:12 2024 -0800

    [RISCV] Add OFFLINE_ASM_ALIGN_TRAP implementation for RISCV
    https://bugs.webkit.org/show_bug.cgi?id=282897
    
    Reviewed by Yusuke Suzuki.
    
    * Source/JavaScriptCore/llint/LowLevelInterpreter.cpp:
    
    Canonical link: https://commits.webkit.org/286407@main

Index: Source/JavaScriptCore/llint/LowLevelInterpreter.cpp
--- Source/JavaScriptCore/llint/LowLevelInterpreter.cpp.orig
+++ Source/JavaScriptCore/llint/LowLevelInterpreter.cpp
@@ -584,6 +584,8 @@ JSValue CLoop::execute(OpcodeID entryOpcodeID, void* e
 #define OFFLINE_ASM_ALIGN_TRAP(align) OFFLINE_ASM_BEGIN_SPACER "\n .balign " #align ", 0xcc\n" // pad with int 3 instructions
 #elif CPU(ARM)
 #define OFFLINE_ASM_ALIGN_TRAP(align) OFFLINE_ASM_BEGIN_SPACER "\n .balignw " #align ", 0xde00\n" // pad with udf instructions
+#elif CPU(RISCV64)
+#define OFFLINE_ASM_ALIGN_TRAP(align) OFFLINE_ASM_BEGIN_SPACER "\n .balign " #align ", 0x73\n" // pad with ebreak instructions
 #endif
 
 #define OFFLINE_ASM_EXPORT_SYMBOL(symbol)
